packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -91,6 +190,48 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
   async start() {}
   async stop() {}
 
+  private propagateResponseExtensions(extensions: DataProxyExtensions): void {
+    const tracingConfig = getTracingConfig(this.config.previewFeatures || [])
+
+    if (extensions?.logs?.length) {
+      extensions.logs.forEach((log) => {
+        switch (log.level) {
+          case 'debug':
+          case 'error':
+          case 'trace':
+          case 'warn':
+          case 'info':
+            // TODO these are propagated into the response.errors key
+            break
+          case 'query': {
+            let dbQuery = typeof log.attributes.query === 'string' && log.attributes.query ? log.attributes.query : ''
+
+            if (!tracingConfig.enabled) {
+              // The engine uses tracing to consolidate logs
+              //  - and so we should strip the generated traceparent
+              //  - if tracing is disabled.
+              // Example query: 'SELECT /* traceparent=00-123-0-01 */'
+              const [query] = dbQuery.split('/* traceparent')
+              dbQuery = query
+            }
+
+            this.logEmitter.emit('query', {
+              query: dbQuery,
+              timestamp: log.timestamp,
+              duration: log.attributes.duration_ms,
+              params: log.attributes.params,
```

Comment:
Here `log.attributes.params` is an array, but `params` is a `string` in the user-facing `QueryEvent` type. What do library and binary engines emit as `params` in their log events? Do we need to change this to
```suggestion
              params: JSON.stringify(log.attributes.params),
```
here or is the type wrong for other engine types too?

Reply:
I think is fine either way, to me, knowing that params is an array, it could be treated as such. However, the information is there fore being rendered to the user. I will default to consistency and make it a string.

Reply:
> I think is fine either way, to me, knowing that params is an array, it could be treated as such.

Yeah, I mean, an array is definitely better, but we also can't break the existing API, so we'd have to change it to an array in Prisma 5 for all `Engine` class implementations if the other engines adhere to the contract there. It's a different question though if the type is plainly wrong and the library and binary engines return an array there too (we already had a precedent of having the types completely wrong for some fields for the library engine there in the past — the internal API boundary here is not type safe at all and we also probably lack tests that would check the actual types).

Making this a string is probably the safest option for now.